src/components/Scheme/ContributionRewardExtRewarders/Competition/utils.ts

```diff
@@ -0,0 +1,190 @@
+  return async (dispatch: Redux.Dispatch<any, any>, _getState: () => IRootState) => {
+    try {
+      const observer = operationNotifierObserver(dispatch, "Vote Submission");
+      const submission = new CompetitionSuggestion(options.id, getArc());
+
+      await submission.vote().subscribe(...observer);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      throw err;
+    }
+  };
+};
+
+export interface IVoteSubmissionOptions {
+  id: string; // actual id, not the counter
+}
+
+export const redeemForSubmission = (options: IVoteSubmissionOptions ): ThunkAction<any, IRootState, null> => {
+  return async (dispatch: Redux.Dispatch<any, any>, _getState: () => IRootState) => {
+    try {
+      const observer = operationNotifierObserver(dispatch, "Redeem Submission");
+      const submission = new CompetitionSuggestion(options.id, getArc());
+
+      await submission.redeem().subscribe(...observer);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      throw err;
+    }
+  };
+};
+
+
+/**
+ * must be an exact subset of ICompetitionSuggestionQueryOptions
+ */
+export interface IGetSubmissionsOptions {
+  id?: string; // id of the competition
+  suggestionId?: string; // the "suggestionId" is a counter that is unique to the scheme
+  // - and is not to be confused with suggestion.id
+}
+
+const getSubmissions = (
+  proposalId: string,
+  options?: IGetSubmissionsOptions,
+  subscribe = false
+): Observable<Array<ICompetitionSuggestion>> => {
+  // FAKE -- until we have IProposalState.competition.suggestions()
+  const competition = new Competition(proposalId, getArc());
+  // return props.proposalState.competition.suggestions({ where: { proposal: props.proposalState.id }}, { subscribe: true } )
+  return competition.suggestions({ where: options }, { subscribe, fetchAllData: true })
+    .pipe(
+      // FAKE -- until .fetchStaticState() exists on CompetitionSuggestion
+      // mergeMap((suggestions: Array<CompetitionSuggestion>) => suggestions.map((suggestion) => from(suggestion.fetchStaticState()) )),
+      // concat()  (??)
+      // or:
+      // map((suggestions: Array<CompetitionSuggestion>) => suggestions.map((suggestion) => suggestion.staticState ))
+
+      // work-around hack because CompetitionSuggestion actually contains all we need
+      map((suggestions: Array<CompetitionSuggestion>) => suggestions.map((suggestion) => suggestion as unknown as ICompetitionSuggestion) )
+    );
+};
+
+export const getProposalSubmissions = (proposalId: string, subscribe = false): Observable<Array<ICompetitionSuggestion>> => {
+  return getSubmissions(proposalId, undefined, subscribe);
+};
+
+export const getProposalSubmission = (proposalId: string, id: string, subscribe = false): Observable<ICompetitionSuggestion> => {
+  /** 
+   * won't subscribe here because the only use case is SubmissionDetails where we can assume the suggestion has
```

Comment:
won't subscribe, or does it dpeend on the parameter?

Reply:
Sorry, I don't understand your question

Reply:
I think this comment may be obsolete